src/Dumper.cpp

```diff
@@ -444,9 +489,40 @@ void Dumper::dumpValue(Slice const* slice, Slice const* base) {
       break;
     }
 
-    case ValueType::UTCDate: 
+    case ValueType::Tagged: {
+      dump(slice->value());
+      break;
+    }
+
+    case ValueType::Binary: {
+      if(options->binaryAsHex) {
+        _sink->push_back('"');
+        ValueLength len;
+	    uint8_t const *bin = slice->getBinary(len);
+        for (uint8_t i = 0; i < len; i++) {
+          uint8_t value = *(bin+i);
+          uint8_t x = value / 16;
+          _sink->push_back((x < 10 ? ('0' + x) : ('a' + x - 10)));
+          x = value % 16;
+          _sink->push_back((x < 10 ? ('0' + x) : ('a' + x - 10)));
+        }
+        _sink->push_back('"');
+      } else {
+        handleUnsupportedType(slice);
+      }
+      break;
+    }
+
+    case ValueType::UTCDate: {
+      if(options->datesAsIntegers) {
+        appendUInt(slice->getUTCDate());
```

Comment:
AFAIK `Slice::getUTCDate()` returns an `int64_t`, so this should be changed to
```suggestion
        appendInt(slice->getUTCDate());
```